src/embed_tests/TestDomainReload.cs

```diff
@@ -50,173 +59,383 @@ class TestDomainReload
-            compilerparams.OutputAssembly = assemblyName;
+                        using (PyObject obj = new PyObject(handle))
+                        {
+                            obj.InvokeMethod("Method");
+                            obj.InvokeMethod("StaticMethod");
 
-            var results = provider.CompileAssemblyFromSource(compilerparams, TestCode);
-            if (results.Errors.HasErrors)
+                            using (var scope = Py.CreateScope())
+                            {
+                                scope.Set("obj", obj);
+                                scope.Exec(@"
+obj.Method()
+obj.StaticMethod()
+obj.Property += 1
+obj.Field += 10
+");
+                            }
+                            var clrObj = obj.As<Domain.MyClass>();
+                            Assert.AreEqual(clrObj.Property, 2);
+                            Assert.AreEqual(clrObj.Field, 20);
+                        }
+                    }
+                }
+                catch (Exception e)
+                {
+                    Debug.WriteLine(e);
+                    throw;
+                }
+                return 0;
+            }
+        }
+
+        [Test]
+        public static void CrossDomainObject()
+        {
+            RunDomainReloadSteps<CrossDomianObjectStep1, CrossDomianObjectStep2>();
+        }
+
+        #endregion
+
+        #region TestClassReference
+
+        class ReloadClassRefStep1 : CrossCaller
+        {
+            public override ValueType Execte(ValueType arg)
             {
-                var errors = new System.Text.StringBuilder("Compiler Errors:\n");
-                foreach (CompilerError error in results.Errors)
+                const string code = @"
+from Python.EmbeddingTest.Domain import MyClass
+
+def test_obj_call():
+    obj = MyClass()
+    obj.Method()
+    obj.StaticMethod()
+    obj.Property = 1
+    obj.Field = 10
+
+test_obj_call()
+";
+                const string name = "test_domain_reload_mod";
+                using (Py.GIL())
                 {
-                    errors.AppendFormat("Line {0},{1}\t: {2}\n",
-                            error.Line, error.Column, error.ErrorText);
+                    IntPtr module = PyRuntime.PyModule_New(name);
+                    Assert.That(module != IntPtr.Zero);
+                    IntPtr globals = PyRuntime.PyObject_GetAttrString(module, "__dict__");
+                    Assert.That(globals != IntPtr.Zero);
+                    try
+                    {
+                        int res = PyRuntime.PyDict_SetItemString(globals, "__builtins__",
+                            PyRuntime.PyEval_GetBuiltins());
+                        PythonException.ThrowIfIsNotZero(res);
+
+                        PythonEngine.Exec(code, globals);
+                        IntPtr modules = PyRuntime.PyImport_GetModuleDict();
+                        res = PyRuntime.PyDict_SetItemString(modules, name, modules);
```

Comment:
It should be `PyDict_SetItemString(modules, name, module);`
Since the clr hold the reference of module, it didn't interfere the test result.
But I should modify the test to clr won't hold reference and use general import operate to get the old module, for better.